tests/ps_test.go

```diff
@@ -1,44 +1,156 @@
+// scrapeProcs returns the sorted process names for an app from the given output.
+// It matches the current "deis ps" output for a healthy container:
+//   earthy-vocalist-v2-cmd-1d73e up (v2)
+//   myapp-v16-web-bujlq up (v16)
+func scrapeProcs(app string, output []byte) []string {
+	re := regexp.MustCompile(fmt.Sprintf(`%s-v\d+-[\w-]+ up \(v\d+\)`, app))
+	found := re.FindAll(output, -1)
+	procs := make([]string, len(found))
+	for i := range found {
+		procs[i] = string(found[i])
+	}
+	sort.Strings(procs)
+	return procs
+}
+
+var _ = Describe("Processes", func() {
+
 	Context("with a deployed app", func() {
 
-		XIt("can scale upward", func() {
-			// "deis ps:scale web=5 --app=%s"
-			// "deis ps:list --app=%s"
-			// curl app
-			// "deis ps:scale web=1 --app=%s"
-			// "deis ps:list --app=%s"
-			// curl app
-		})
+		appName := getRandAppName()
+		appURL := strings.Replace(url, "deis", appName, 1)
+		once := &sync.Once{}
 
-		XIt("can scale down to 0", func() {
-			// "deis ps:scale web=0 --app=%s"
-			// "deis ps:list --app=%s"
-			// curl app
-			// "deis ps:scale web=1--app=%s"
-			// "deis ps:list --app=%s"
-			// curl app
+		BeforeEach(func() {
+			// Set up the Processes test app only once and assume the suite will clean up.
+			once.Do(func() {
+				os.Chdir("example-go")
+				cmd := createApp(appName)
+				Eventually(cmd).Should(SatisfyAll(
+					Say("Git remote deis added"),
+					Say("remote available at ")))
+				Eventually(cmd).Should(Exit(0))
+				cmd, err := start("GIT_SSH=%s git push deis master", gitSSH)
+				Expect(err).NotTo(HaveOccurred())
+				Eventually(cmd.Err, "2m").Should(Say("Done, %s:v2 deployed to Deis", appName))
+				Eventually(cmd).Should(Exit(0))
+			})
 		})
 
-		XIt("can restart all processes", func() {
-			// "deis ps:scale web=5 --app=%s"
-			// "deis ps:list --app=%s"
-			// curl app
-			// "deis ps:restart web --app=%s"
-			// "deis ps:list --app=%s"
-			// curl app
-		})
+		DescribeTable("can scale up and down",
 
-		XIt("can restart a specific process", func() {
-			// "deis ps:restart web.1 --app=%s"
-			// "deis ps:list --app=%s"
-			// curl app
-		})
+			func(scaleTo, respCode int) {
+				// TODO: need some way to choose between "web" and "cmd" here!
+				// scale the app's processes to the desired number
+				sess, err := start("deis ps:scale web=%d --app=%s", scaleTo, appName)
+				Expect(err).NotTo(HaveOccurred())
+				Eventually(sess).Should(Say("Scaling processes... but first,"))
+				Eventually(sess).Should(Say(`done in \d+s`))
+				Eventually(sess).Should(Say("=== %s Processes", appName))
+				Eventually(sess).Should(Exit(0))
+
+				// test that there are the right number of processes listed
+				sess, err = start("deis ps:list --app=%s", appName)
+				Expect(err).NotTo(HaveOccurred())
+				Eventually(sess).Should(Say("=== %s Processes", appName))
+				Eventually(sess).Should(Exit(0))
+				re := regexp.MustCompile(fmt.Sprintf(`%s-v\d-[\w-]+ up \(v\d\)`, appName))
```

Comment:
What is the difference between this `re` and the one in `scrapeProcs` above?  Any chance they can be merged into one standalone, reusable regexp?